### ISSUE TITLE
reconfigure jobs: null pointer exception when view has no description

### ISSUE THREAD

**Issue by nuclearsandwich** [ros-infrastructure/ros_buildfarm#513]
It appears that the null checking in https://github.com/ros-infrastructure/ros_buildfarm/blob/93d19cb8003ecb34317c63c0ad0284973ddb04c3/ros_buildfarm/templates/snippet/reconfigure_jobs.groovy.em#L250-L256 appears to be insufficient. Somehow the Mdev_ds views were created without descriptions and later reconfigure jobs failed with null pointer references when running. An example build: http://build.ros.org/job/Mdev_ds_reconfigure-jobs/2

Build output:

<details>
```
# BEGIN SUBSECTION: reconfigure 2 views
13:12:47 ERROR: Build step failed with exception
13:12:47 java.lang.NullPointerException: Cannot invoke method setTextContent() on null object
13:12:47 	at org.codehaus.groovy.runtime.NullObject.invokeMethod(NullObject.java:91)
13:12:47 	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:48)
13:12:47 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
13:12:47 	at org.codehaus.groovy.runtime.callsite.NullCallSite.call(NullCallSite.java:35)
13:12:47 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
13:12:47 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
13:12:47 	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:155)
13:12:47 	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:157)
13:12:47 	at org.kohsuke.groovy.sandbox.impl.Checker$checkedCall.callStatic(Unknown Source)
13:12:47 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:56)
13:12:47 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:194)
13:12:47 	at Script1.diff_configs(Script1.groovy:231)
13:12:47 	at Script1$diff_configs.call(Unknown Source)
13:12:47 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
13:12:47 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
13:12:47 	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:155)
13:12:47 	at org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:23)
13:12:47 	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:133)
13:12:47 	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:153)
13:12:47 	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:157)
13:12:47 	at org.kohsuke.groovy.sandbox.impl.Checker$checkedCall.callStatic(Unknown Source)
13:12:47 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:56)
13:12:47 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:194)
13:12:47 	at Script1.run(Script1.groovy:56)
13:12:47 	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.run(GroovySandbox.java:141)
13:12:47 	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SecureGroovyScript.evaluate(SecureGroovyScript.java:165)
13:12:47 	at hudson.plugins.groovy.SystemGroovy.run(SystemGroovy.java:95)
13:12:47 	at hudson.plugins.groovy.SystemGroovy.perform(SystemGroovy.java:59)
13:12:47 	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
13:12:47 	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:735)
13:12:47 	at hudson.model.Build$BuildExecution.build(Build.java:206)
13:12:47 	at hudson.model.Build$BuildExecution.doRun(Build.java:163)
13:12:47 	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:490)
13:12:47 	at hudson.model.Run.execute(Run.java:1735)
13:12:47 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
13:12:47 	at hudson.model.ResourceController.execute(ResourceController.java:97)
13:12:47 	at hudson.model.Executor.run(Executor.java:405)
13:12:47 Build step 'Execute system Groovy script' marked build as failure
13:12:47 Sending e-mails to: clalancette+buildfarm@osrfoundation.org ros-buildfarm-melodic@googlegroups.com
13:12:47 Finished: FAILURE
```
</details>

cc @mikaelarguedas

**Comment by dirk-thomas** [comment]
* Does that happen when the views are created from scratch using the maintenance jobs? In this case I think they were created by a script run locally.

* Why did the following build no 3 succeed? What has changed in between these two builds?

**Comment by nuclearsandwich** [comment]
> Does that happen when the views are created from scratch using the maintenance jobs? In this case I think they were created by a script run locally.

I don't know the answer to this question. Perhaps @mikaelarguedas does.

> Why did the following build no 3 succeed? What has changed in between these two builds?

I added the description `asdf` to both views by hand before re-triggering it.

**Comment by mikaelarguedas** [comment]
> > Does that happen when the views are created from scratch using the maintenance jobs? In this case I think they were created by a script run locally.

> I don't know the answer to this question. Perhaps @mikaelarguedas does.

Indeed the (empty/failing) views were created locally with a script. I don't think the problem happens when they are created using the maintenance jobs as we didn't face any issue when the melodic related views were created though I haven't tested it.

**Comment by dirk-thomas** [comment]
Addressed by #518.
